### PR TITLE
Finally... Gamification! (For real!)

### PR DESCRIPTION
**Pull request by bramirez96**
# Description

This PR includes data models and endpoints for the rest of the gamification process:

- [x] Pull team and submission data for meeting partners/point allocation on Wednesdays
- [x] Send point allocations to the database after children have made their decisions
- [x] Generate matchups after points have been allocated
- [x] Return matchup information for a squad
- [x] Submit votes for different matchups
- [x] Calculate the winning team based on points and votes
- [x] Send results of the week's game for children to see

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: __tests__/routes/game-test.js [hunk 1]
```diff
@@ -115,9 +115,374 @@ module.exports = () => {
...
+      let res1;
+      it('sends submissions for a child and their teammate', async () => {
+        const res = await request(server).get(
+          `/game/team?childId=${childIds[0]}`
+        );
+
+        expect(res.status).toBe(200);
+        expect(Object.keys(res.body)).toHaveLength(3);
+        res1 = res.body;
+        // console.log({ getTeam: res.body, pages: res.body[1].Pages });
```
Comment (Schrese) [inline __tests__/routes/game-test.js]: might need to remove this console.log

File: api/game/gameHelpers/faceoffRetrieval.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const { dbOps, formatSubForMatchups } = require('../../../lib');
+
```
Comment (Schrese) [inline api/game/gameHelpers/faceoffRetrieval.js:2]: missing jsDocs?

**Review by SandraCoburn — APPROVED** [review]
This is great!

File: api/mod/modHelpers/clusterGeneration.js [hunk 3]
```diff
@@ -21,24 +11,28 @@ const setComplexity = (ID, Complexity) => {
...
 
       // Add the generated clusters to the database
+      let members = [];
       for (let { ID } of cohorts) {
         for (let squad of clusters[ID]) {
           const [SquadID] = await addSquad(trx, ID);
           const [t1, t2] = await addTeams(trx, SquadID);
-          await addMembers(trx, t1, t2, squad);
+          const newMembers = await addMembers(trx, t1, t2, squad);
+          members.push([].concat.apply([], newMembers));
```
Comment (Schrese) [inline api/mod/modHelpers/clusterGeneration.js:29]: This is a cool line for concatenations

File: api/mod/modHelpers/resultCalulation.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+ * @returns {Promise} returns an object with { Points, Winner, SquadID }
+ */
+const countVotes = async (conn) => {
+  // Pull all votes from the database and initialize results object
+  const votes = await getVotingResults(conn);
+  const res = {};
+  // Iterate over all of the votes from the database
+  votes.forEach(({ Vote, ID, Points, SquadID }) => {
+    // Initialize vote counters for team 1 and 2
+    if (!res[ID]) res[ID] = { 1: 0, 2: 0 };
```
Comment (Schrese) [inline api/mod/modHelpers/resultCalulation.js:17]: This is cool

File: api/mod/modHelpers/resultCalulation.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+ * @param {number} ID the unique integer ID of the squad
+ * @param {number} Num the team number, either 1 or 2
+ * @param {number} Points the number of points earned by the team
+ * @returns {Promise}
+ */
+const updatePointsForTeam = (conn, ID, Num, Points) => {
+  return conn('Teams AS T')
+    .join('Squads AS S', 'S.ID', 'T.SquadID')
+    .where('S.ID', ID)
+    .andWhere('T.Num', Num)
```
Comment (Schrese) [inline api/mod/modHelpers/resultCalulation.js:130]: Didn't know that this existed

File: lib/errorHandler.js [hunk 6]
```diff
@@ -32,6 +32,9 @@ const send = (res, error, name = '') => {
   } else if (/column.+of relation.+does not exist/g.test(error.message)) {
```
Comment (Schrese) [inline lib/errorHandler.js:32]: this regex is super cool

File: __tests__/routes/game-test.js (hunk 1 above)
Comment (bramirez96) [reply __tests__/routes/game-test.js]: done!

**Review by Schrese — APPROVED** [review]

**Pull request merged**